scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -468,58 +515,68 @@ We can correct this by applying a "shrinkage" procedure, which will adjust large
 
 To do this, we will use the `lfcShrink()` function, but first we need to know the name and/or position of the "coefficient" that was calculated by `DESeq()`, which we can do with the `resultsNames()` function.
 
-```{r}
+```{r coefficient}
 # identify position of coefficient
 DESeq2::resultsNames(deseq_object)
 ```
 
 
-```{r}
-# appyly logFC shrinkage
+```{r shrinkage}
+# appyly logFC shrinkage using the default model
 shrink_results <- DESeq2::lfcShrink(deseq_object, res = deseq_results, coef = 2, type = "apeglm")
+head(shrink_results)
 ```
 
-Now we can get the updated results and combine with the `rowData` from the SCE object to extract the gene symbol. 
+If you look at our `shrink_results` object, we see that the genes are labeled with the ensembl gene identifiers, as those were the rownames of the pseudo-bulked SCE we used as input to build our `DESeq2` object.
+Although some of us may have all of the identifiers memorized by heart, it can be useful to have a human readable symbol in our results.
+Before we save the results as a file, we will grab the gene symbols from the `rowData` of our original SCE object and add them as a new column.
 
-```{r}
-# create a data frame with the results 
-deseq_results <- shrink_results %>%
+```{r add gene symbol}
+# convert shrunken results to a data.frame and add ensembl id as a column
+deseq_results <- shrink_results |>
   tibble::as_tibble(rownames = "ensembl_id")
 
-# first look at the significant results 
-deseq_results_sig <- deseq_results %>%
-  dplyr::filter(padj <= 0.05)
-
-deseq_results_sig
-```
-
-
-```{r}
-# convert rowdata to dataframe 
-sce_rowdata_df <- rowData(tumor_myoblast_sce) %>%
+# convert rowdata to data.frame 
+sce_rowdata_df <- rowData(tumor_myoblast_sce) |>
   # create a column with rownames stored as ensembl id to use for joining with deseq results
   tibble::as_tibble(rownames = "ensembl_id")
 
 # combine deseq results with rowdata by ensembl id 
-deseq_results <- deseq_results %>%
+deseq_results <- deseq_results |>
   dplyr::left_join(sce_rowdata_df, by = "ensembl_id")
 
 head(deseq_results)
 ```
 
-```{r}
-# save our results 
+We can save the new `data.frame` that we have created with the ensembl identifiers, gene symbols, and the `DESeq2` results as a tab separated (`tsv`) file.
```

Comment:
```suggestion
We can save the new `data.frame` that we have created with the Ensembl identifiers, gene symbols, and the `DESeq2` results as a tab separated (`tsv`) file.
```

Reply:
I think I might remove `data.frame` --> "data frame" since this isn't referencing that function, just saying the noun "data frame."
Though there are so many kinds of data frames out there that there is a good argument for not doing that!